Description: Detect KFreeBSD (simply add it whereever FreeBSD is listed)
Author: Jonas Smedegaard <dr@jones.dk>
Last-Update: 2013-09-25

--- a/uwsgiconfig.py
+++ b/uwsgiconfig.py
@@ -655,7 +655,7 @@
 
         global uwsgi_version
 
-        kvm_list = ['FreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly']
+        kvm_list = ['FreeBSD', 'KFreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly']
 
         if 'UWSGI_AS_LIB' in os.environ:
             self.set('as_shared_library', 'true')
@@ -665,7 +665,7 @@
             self.cflags.append('-DUWSGI_HAS_IFADDRS')
             report['ifaddrs'] = True
 
-        if uwsgi_os in ('FreeBSD', 'OpenBSD'):
+        if uwsgi_os in ('FreeBSD', 'KFreeBSD', 'OpenBSD'):
             if self.has_include('execinfo.h') or os.path.exists('/usr/local/include/execinfo.h'):
                 if os.path.exists('/usr/local/include/execinfo.h'):
                     self.cflags.append('-I/usr/local/include')
@@ -730,7 +730,7 @@
                 locking_mode = 'pthread_mutex'
             # FreeBSD umtx is still not ready for process shared locking
             # starting from FreeBSD 9 posix semaphores can be shared between processes
-            elif uwsgi_os == 'FreeBSD':
+            elif uwsgi_os == 'FreeBSD' or uwsgi_os == 'KFreeBSD':
                  try:
                      fbsd_major = int(uwsgi_os_k.split('.')[0])
                      if fbsd_major >= 9:
@@ -773,7 +773,7 @@
                 if int(sun_major) >= 5:
                     if int(sun_minor) >= 10:
                         event_mode = 'port'
-            elif uwsgi_os in ('Darwin', 'FreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly'):
+            elif uwsgi_os in ('Darwin', 'FreeBSD', 'KFreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly'):
                 event_mode = 'kqueue'
             elif uwsgi_os.startswith('CYGWIN') or uwsgi_os == 'GNU':
                 event_mode = 'poll'
@@ -816,7 +816,7 @@
                     if int(sun_minor) >= 10:
                         timer_mode = 'port'
 
-            elif uwsgi_os in ('Darwin', 'FreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly'):
+            elif uwsgi_os in ('Darwin', 'FreeBSD', 'KFreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly'):
                 timer_mode = 'kqueue'
 
         if timer_mode == 'timerfd':
@@ -843,7 +843,7 @@
                 if int(sun_major) >= 5:
                     if int(sun_minor) >= 10:
                         filemonitor_mode = 'port'
-            elif uwsgi_os in ('Darwin', 'FreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly'):
+            elif uwsgi_os in ('Darwin', 'FreeBSD', 'KFreeBSD', 'OpenBSD', 'NetBSD', 'DragonFly'):
                 filemonitor_mode = 'kqueue'
 
         if filemonitor_mode == 'inotify':
@@ -952,7 +952,7 @@
             uwsgi_version += self.get('append_version')
 
 
-        if uwsgi_os == 'FreeBSD' and self.has_include('jail.h'):
+        if (uwsgi_os == 'FreeBSD' or uwsgi_os == 'KFreeBSD') and self.has_include('jail.h'):
             self.cflags.append('-DUWSGI_HAS_FREEBSD_LIBJAIL')
             self.libs.append('-ljail')
 
